test/ConnectionFactoryTest.php

```diff
@@ -70,13 +82,9 @@ function (string $id) {
             foreach ($e->getTrace() as $entry) {
                 if (
                     isset($entry['class'])
-                    && in_array($entry['class'], [
-                        PDOMySQLDriver::class,
-                        AbstractMySQLDriver::class,
-                        ExceptionConverter::class,
```

Comment:
Wait... `ExceptionConverter` is gone on purpose? :thinking:

Reply:
Conditional, not sure if its only available in lower versions or higher versions. would wonder if it exists only in lower versions 🤷🏼‍♂️

Reply:
@boesing could you raise the minimum version then (if not already the case)?

Reply:
Sure, I could do that, but I would prefer not to do so due to the fact that it is not really necessary for the coverage of this project.

### TL;DR

- the `ExceptionConverter` was introduced in `doctrine/dbal` v3.0.0
- dropping support for `doctrine/dbal` v2 would also contain code cleanup and additions to this repo which I would prefer not to do in a `vimeo/psalm` update

---

So I could also just bump the `composer.lock` to contain `doctrine/dbal` `^3.0` and psalm won't complain anymore.
Easier change, less impact, same result.

WDYT?

Reply:
Okay, I'll stop working on this and focus on the PSR-6 support.
After this change, some deprecation quirks can be removed and its easier to drop support for `doctrine/dbal` <3

So I will remove the psalm changes from that upcoming PR to then rebase this PR off that PR.

--- 

But just to leave this here: 

this is kinda annoying and absolutely unnecessary from my current PoV. 
From a functionality aspect, this just takes more of my time than necessary. Without this PR, this won't be a problem and I doubt that it would change anytime soon - but now, as there are changes and you've seen it, you want me to put more effort in this. 

But I will do as you wish as this is your repository and therefore your rules. 

> Nope, either the code is valid for the ranges that we support, or it isn't :P

And it is. Thats why tests are passing on lowest and latest. Not sure what you want to say here but as I said: your repository, your rules.